lib/setup-cookie-recording.js

```diff
@@ -129,30 +129,40 @@ module.exports.setup_cookie_recording = async function(page) {
         fileName: req.url(),
         source: `set in Set-Cookie HTTP response header for ${req.url()}`,
       }];
-      let splitCookieHeaders = cookieHTTP.split("\n");
-      let data = splitCookieHeaders.map(cookieParser.parse);
-      data.forEach( (cookie) => {
-        if(!cookie.domain) {
+      const domain = new url.URL(response.url()).hostname;
+      const data = cookieHTTP.split("\n")
+        .map(c => { return cookieParser.parse(c) || {value: c}; })
+        .map(cookie => {
           // what is the domain if not set explicitly?
           // https://stackoverflow.com/a/5258477/1407622
-          cookie.domain = new url.URL(response.url()).hostname;
-        }
-      });
-      let message = `${data.length} Cookie(s) (HTTP) set for host ${data[0].domain} with key(s) ${data.map(c => {return c.key;}).join(', ')}.`;
+          cookie.domain ||= domain;
+          return cookie;
+        });
+
+      const { valid, invalid } = data.reduce((a, c) => {
+        return c.key ? Object.assign(a, { valid: a.valid.concat(c) }) : Object.assign(a, { invalid: a.invalid.concat(c)});
+      }, { valid:[], invalid:[] });
```

Comment:
I see you are a JS ninja and can group the cookies without further dependencies. For the sake of readability, I would use lodash here.

```suggestion
      const dataHasKey = groupBy(data, (cookie) => {return !!cookie.key;});
      const valid = dataHashKey[true];
      const invalid = dataHashKey[false];
```

Not sure if valid and invalid as separate variables are even necessary then.

Reply:
Actually, I'm still learning JS and try to avoid helper libraries at this point until I feel I have a full grasp of things :)
But I'm fine with whatever works. If you feel that the lodash makes things clearer, I'm all for it.